risc0/zkvm/src/host/server/exec/syscall/args.rs

```diff
@@ -50,6 +55,11 @@ impl Syscall for SysArgs {
             let nbytes = min(to_guest.len() * WORD_SIZE, arg_val.len());
             let to_guest_u8s: &mut [u8] = bytemuck::cast_slice_mut(to_guest);
             to_guest_u8s[0..nbytes].clone_from_slice(&arg_val.as_bytes()[0..nbytes]);
+
+            if arg_val.len() == MAX_IO_BYTES as usize {
```

Comment:
```suggestion
            if arg_val.len() >= MAX_IO_BYTES as usize {
```
I think you meant this. There's a few spots where you have `==` instead of `>=`